<?xml version="1.0" ?>
<launch>
  <arg name="load_cam" default="false"/>
  <arg name="load_bag" default="false"/>
  <arg name="rviz" default="false"/>
  <arg name="log_debug" default="false"/>

  <node name="object_keypoints_ros" pkg="object_keypoints_ros" type="run_pipeline.py" output="screen">
    <param name="image_topic" value="/camera/color/image_raw" />
    <param name="camera_frame" value="camera_color_optical_frame" />
    <param name="load_model" value="/home/user/perception/model/model_v28.pt" />
    <param name="calibration" value="$(find object_keypoints_ros)/config/calibration.yaml" />
    <param name="keypoints" value="$(find object_keypoints_ros)/config/cups.json" />
    <param name="gpu" value="true" type="bool" />
    <param name="log_debug" value="$(arg log_debug)" type ="bool" />
    <param name="ref_gt" value="/home/user/perception/dataset/encoded/train_large/encode_clutter_env/2000/keypoints.json"/>
  </node>

  <!-- add kp markers topic "kp_3d_marker" for rviz -->>
  <node name="kp_msg_vis_node" pkg="object_keypoints_ros" type="msg_visualization.py" output="screen"/>

  <group if="$(arg load_cam)">
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" />
  </group>
  
  <node pkg="rosbag" type="play" name="rosbag_play_recoreded" args=" -r 0.1 /home/boyang/data/cups/000.bag" if="$(arg load_bag)" >
    <param name="depth_image_topic" value="/camera/aligned_depth_to_color/image_raw"/>
  </node>

  <group if="$(arg rviz)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find object_keypoints_ros)/config/vis_1.rviz" output="screen"/>
  </group>

</launch>
